x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -271,7 +282,32 @@ public void onIndexModule(IndexModule indexModule) {
         if (SearchableSnapshotsConstants.isSearchableSnapshotStore(indexSettings.getSettings())
             && indexSettings.getSettings().getAsBoolean("index.frozen", false) == false) {
             return Optional.of(
-                engineConfig -> new ReadOnlyEngine(engineConfig, null, new TranslogStats(), false, Function.identity(), false)
+                engineConfig -> new ReadOnlyEngine(engineConfig, null, new TranslogStats(), false, Function.identity(), false) {
+
+                    // present an empty IndexCommit to the snapshot mechanism so that we copy no shard data to the repository
+                    private final IndexCommit emptyIndexCommit;
+
+                    {
+                        try {
+                            final Directory directory = engineConfig.getStore().directory();
+                            final String oldestSegmentsFile = Arrays.stream(directory.listAll())
+                                .filter(s -> s.startsWith(IndexFileNames.SEGMENTS + "_"))
+                                .min(Comparator.naturalOrder())
+                                .orElseThrow(() -> new IOException("segments_N file not found"));
+                            final SegmentInfos segmentInfos = new SegmentInfos(Version.LATEST.major);
+                            segmentInfos.updateGeneration(SegmentInfos.readCommit(directory, oldestSegmentsFile));
+                            emptyIndexCommit = Lucene.getIndexCommit(segmentInfos, directory);
```

Comment:
Just for my own learning probably, but why do we need this? Can't we just always use generation `0`? Probably not :) => can we add a comment with why we need this?

Reply:
It's a good question and what I spent most of my time on TBH, I added a comment above.